test/livebook/hubs/fly_client_test.exs

```diff
@@ -68,4 +68,42 @@ defmodule Livebook.Hubs.FlyClientTest do
       assert {:error, "request failed with code: UNAUTHORIZED"} = FlyClient.fetch_apps("foo")
     end
   end
+
+  describe "fetch_app/1" do
+    test "fetches an application", %{bypass: bypass} do
+      app = %{
+        "id" => "foo-app",
+        "name" => "foo-app",
+        "hostname" => "foo-app.fly.dev",
+        "platformVersion" => "nomad",
+        "deployed" => true,
+        "status" => "running"
+      }
+
+      response = %{"data" => %{"app" => app}}
+
+      Bypass.expect_once(bypass, "POST", "/", fn conn ->
+        conn
+        |> Plug.Conn.put_resp_content_type("application/json")
+        |> Plug.Conn.resp(200, Jason.encode!(response))
+      end)
+
+      hub = build(:fly)
+      assert {:ok, ^app} = FlyClient.fetch_app(hub)
+    end
+
+    test "returns unauthorized when token is invalid", %{bypass: bypass} do
+      error = %{"extensions" => %{"code" => "UNAUTHORIZED"}}
+      response = %{"data" => nil, "errors" => [error]}
+
+      Bypass.expect_once(bypass, "POST", "/", fn conn ->
+        conn
+        |> Plug.Conn.put_resp_content_type("application/json")
+        |> Plug.Conn.resp(200, Jason.encode!(response))
+      end)
```

Comment:
Maybe one day we could use Req. ;)

```diff
-  setup do
-    bypass = Bypass.open()
-
-    Application.put_env(
-      :livebook,
-      :fly_graphql_endpoint,
-      "http://localhost:#{bypass.port}"
-    )
-
-    on_exit(fn ->
-      Application.delete_env(:livebook, :fly_graphql_endpoint)
-    end)
-
-    {:ok, bypass: bypass, url: "http://localhost:#{bypass.port}"}
-  end
-
-  ...
-
-  Bypass.expect_once(bypass, "POST", "/", fn conn ->
-    conn
-    |> Plug.Conn.put_resp_content_type("application/json")
-    |> Plug.Conn.resp(200, Jason.encode!(response))
-  end)
-
-  FlyClient.fetch_app(hub)
-
+  adapter = fn request ->
+    {request, Req.Response.json(response)}
+  end)
+
+  FlyClient.fetch_app(hub, adapter: adapter)
```

Reply:
In this case, we would need to change the `S3` file system to use Req too, right? Coz I'm using the same HTTP module which uses the native `:httpc`.

Reply:
Yeah let's not change anything anytime soon. I just meant it would be kinda nice, wouldn't it? Maybe I shouldn't have said anything. :)

Reply:
I wanted to use Req in the first place, but since we already have the `:httpc` I thought it was because of `cli` version